test/e2e/gateway/009_tls_wildcard_host_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build e2e
+
+package gateway
+
+import (
+	"crypto/tls"
+
+	gatewayv1alpha1 "sigs.k8s.io/gateway-api/apis/v1alpha1"
+
+	"github.com/projectcontour/contour/test/e2e"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func testTLSWildcardHost(fx *e2e.Framework) {
+	t := fx.T()
+	namespace := "009-gateway-tls-wildcard-host"
+	hostSuffix := "wildcardhost.gateway.projectcontour.io"
+
+	fx.CreateNamespace(namespace)
+	defer fx.DeleteNamespace(namespace)
+
+	fx.Fixtures.Echo.Deploy(namespace, "echo")
+	fx.Certs.CreateSelfSignedCert(namespace, "echo-one-cert", "echo-one-cert", "*."+hostSuffix)
```

Comment:
Does this cert need to be configured on the Gateway? If so, see https://github.com/projectcontour/contour/blob/5e45ddc6a35b95810d5d3e86f3dd91d83f10ccf0/test/e2e/gateway/gateway_test.go#L144-L150 and https://github.com/projectcontour/contour/blob/5e45ddc6a35b95810d5d3e86f3dd91d83f10ccf0/test/e2e/gateway/gateway_test.go#L166 for how that's done. You'd need a new [Describe](https://github.com/projectcontour/contour/blob/5e45ddc6a35b95810d5d3e86f3dd91d83f10ccf0/test/e2e/gateway/gateway_test.go#L113) block for the Gateway with the wildcard cert